sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/parquet/ParquetRowConverter.scala

```diff
@@ -202,11 +204,15 @@ private[parquet] class ParquetRowConverter(
 
   override def start(): Unit = {
     var i = 0
-    while (i < currentRow.numFields) {
+    while (i < fieldConverters.length) {
       fieldConverters(i).updater.start()
       currentRow.setNullAt(i)
       i += 1
     }
+    while (i < currentRow.numFields) {
```

Comment:
@mallman It sounds like the changes in this file are not needed. Could you help me point out which test cases will fail?

Reply:
I'll get back to you on this shortly. Thanks.

Reply:
These changes are related to my fix for the ignored unit test. If I apply my fix but keep the master version of this file, 24 unit tests fail. If I apply my fix along with this file diff then all tests pass, including the test that is currently ignored.

I'm not sure I can develop a unit test for this current commit that should pass but will fail without this file's changes. I haven't spent any time thinking about it, and I really need to work on other things right now.

If you want I will back out this change. However, I will re-incorporate it in a follow-on PR.

Thanks.

Reply:
My suggestion is to remove the changes from this file at first. We can handle the ignored cases in the follow-up PRs.